File: src/main/java/eu/caple/cipster/checkers/domain/model/Move.java
package eu.caple.cipster.checkers.domain.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.validation.constraints.NotNull;

@Data
public class Move {

	@NotNull
	private Disk disk;

	@EqualsAndHashCode.Exclude
	private Disk capturedDisk;
	private Type type;

	@EqualsAndHashCode.Exclude
	@NotNull
	private Position from;

	@NotNull
	private Position destination;

	public enum Type {
		SLIDE,
		JUMP
	}
}

File: frontend/src/app/checkers/domain/move.ts
import {Disk} from "./disk";
import {Position} from "./position";

export declare type MOVE_TYPE = 'SLIDE' | 'JUMP';

export interface Move {
  disk: Disk;
  capturedDisk?: Disk;
  type?: MOVE_TYPE;
  from: Position;
  destination: Position;
}
